Update DG section 1.5 Overview of the Application (#329)
AY2122S1-CS2103T-W16-1/tp: docs/DeveloperGuide.md

### 1.5 Overview of Application

> In order to better understand how DonnaFin helps financial advisors, it is useful to understand and model the status
> quo without DonnaFin.

Reviewer: ```suggestion

In order to better understand how DonnaFin helps financial advisors, it is useful to understand and model the
key client details that a financial advisor has to keep track of.
```

> Below is an Object-Oriented Domain Model (OODM) modelling the existing state of affairs between a Financial Advisor and
> their Client/s.

Reviewer: ```suggestion
Below is an Object-Oriented Domain Model (OODM) modelling the existing state of affairs between a financial advisor and
their client/s.
```
This would be a noun, not a proper noun or a class name.

> ![OverviewOodmDiagram](images/OverviewOodmDiagram.png)

Reviewer: ```suggestion
![Overview OODM Diagram](images/OverviewOodmDiagram.png)
```
the alt text is normal text usually, no need to PascalCase it.

(alt text is the portion between the `[]`)

Reviewer 2: OODM in caps!

We can see that Financial Advisors generally have one or more clients that they have to manage, and
every client has the following unique information that is important for the Financial Advisors to keep track of:
take note of:
1. `Contact Details`
2. `Policies`
3. `Assets`
4. `Liabilities`

> 5. `Notes`

Reviewer: * Financial Advisors are not proper nouns, should be all lower case
* I believe the client has the "following categories of information unique to them" is more appropriate to these listed categories of info.
* the listed categories of information should not be in code form, just normal text as they are normal, non-code category names.

From the diagram above it can be seen that there is no solution for the Financial Advisors to keep track of all these
details, and this is where DonnaFin comes in.

DonnaFin gives Financial Advisors a platform to multiple clients, and store the `Contact Details`, `Policies`, `Assets`,
`Liabilities`, and `Notes` for each and every client they want to register with the application. In the Developer Guide
it will be explained how all the different classes, and components of DonnaFin come together to create a software that

> is able to perform this functionality.

Reviewer: Nicely phrased. Note that it should not be code fenced though.
